src/System.CommandLine/ArgumentResult.cs

```diff
@@ -5,14 +5,19 @@ namespace System.CommandLine
 {
     public abstract class ArgumentResult
     {
-        private protected ArgumentResult()
+        protected internal ArgumentResult(IArgument argument)
         {
+            Argument = argument;
         }
 
-        public static FailedArgumentResult Failure(string error) => new FailedArgumentResult(error);
+        public IArgument Argument { get; }
 
-        public static SuccessfulArgumentResult Success(object value) => new SuccessfulArgumentResult(value);
+        public string ErrorMessage { get; set; }
```

Comment:
It seems un-usual to have an ErrorMessage on the base class. Can this be moved to the `FailedArgumentResult` class?

Reply:
It's pretty weird. The whole class has to be replaced. I can make this internal in the meantime.